File: 1457_Pseudo-Palindromic_Paths_in_a_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    int traverse (TreeNode* node, int count) {
        if (!node) return 0;
        count ^= 1 << (node->val - 1);
        int res = traverse(node->left, count) + traverse(node->right, count);
        if (!node->left && !node->right && (count & (count - 1)) == 0) res++;
        return res;
    }

public:
    int pseudoPalindromicPaths (TreeNode* root) {
        return traverse(root, 0);
    }
};

File: 1457_Pseudo-Palindromic_Paths_in_a_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pseudoPalindromicPaths (self, root: Optional[TreeNode]) -> int:
        def traverse(node, record):
            if not node:
                return 0
            
            record[node.val - 1] += 1
            if not node.left and not node.right:
                odd = 0
                for r in record:
                    odd += r % 2
                return 1 if odd <= 1 else 0
            return traverse(node.left, record.copy()) + traverse(node.right, record.copy())
        
        return traverse(root, [0 for i in range(9)])
